Sources/_CryptoExtras/MLDSA/MLDSA_boring.swift

```diff
@@ -0,0 +1,319 @@
+        public init() throws {
+            self.pointer = UnsafeMutablePointer<MLDSA65_private_key>.allocate(capacity: 1)
+
+            let publicKeyPtr = UnsafeMutablePointer<UInt8>.allocate(capacity: MLDSA.PublicKey.bytesCount)
+            let seedPtr = UnsafeMutablePointer<UInt8>.allocate(capacity: MLDSA.seedSizeInBytes)
+            defer {
+                publicKeyPtr.deallocate()
+                seedPtr.deallocate()
+            }
+
+            guard CCryptoBoringSSL_MLDSA65_generate_key(
+                publicKeyPtr,
+                seedPtr,
+                self.pointer
+            ) == 1 else {
+                throw CryptoMLDSAError.keyGenerationFailure
+            }
+        }
+
+        /// Initialize a ML-DSA-65 private key from a seed.
+        /// 
+        /// The seed must be at least 32 bytes long.
+        /// Any additional bytes in the seed are ignored.
+        /// 
+        /// - Parameter seed: The seed to use to generate the private key.
+        /// 
+        /// - Throws: `CryptoKitError.incorrectKeySize` if the seed is not at least 32 bytes long or ``CryptoMLDSAError/keyGenerationFailure`` if the key could not be generated.
+        public init(from seed: some DataProtocol) throws {
+            guard seed.count >= MLDSA.seedSizeInBytes else {
+                throw CryptoKitError.incorrectKeySize
+            }
+
+            let seedPtr = UnsafeMutablePointer<UInt8>.allocate(capacity: MLDSA.seedSizeInBytes)
+            seedPtr.initialize(from: seed.regions.flatMap { $0 }, count: MLDSA.seedSizeInBytes)
+            defer { seedPtr.deallocate() }
+
+            self.pointer = UnsafeMutablePointer<MLDSA65_private_key>.allocate(capacity: 1)
+
+            guard CCryptoBoringSSL_MLDSA65_private_key_from_seed(
+                self.pointer,
+                seedPtr,
+                MLDSA.seedSizeInBytes
+            ) == 1 else {
+                throw CryptoMLDSAError.keyGenerationFailure
+            }
+        }
+
+        /// Initialize a ML-DSA-65 private key from a DER representation.
+        /// 
+        /// - Parameter derRepresentation: The DER representation of the private key.
+        /// 
+        /// - Throws: `CryptoKitError.incorrectKeySize` if the DER representation is not the correct size or ``CryptoMLDSAError/keyGenerationFailure`` if the key could not be generated.
+        public init(derRepresentation: some DataProtocol) throws {
+            guard derRepresentation.count == MLDSA.PrivateKey.bytesCount else {
+                throw CryptoKitError.incorrectKeySize
+            }
+
+            self.pointer = UnsafeMutablePointer<MLDSA65_private_key>.allocate(capacity: 1)
+
+            let cbsPointer = UnsafeMutablePointer<CBS>.allocate(capacity: 1)
+            defer { cbsPointer.deallocate() }
+            derRepresentation.regions.flatMap { $0 }.withUnsafeBufferPointer { buffer in
+                cbsPointer.pointee = CBS(data: buffer.baseAddress, len: buffer.count)
```

Comment:
This is unsound: it escapes `buffer` from the closure, which is forbidden. You must use the CBS entirely within this closure.